src-tauri/src/constructors/init/init_edge_map.rs

```diff
@@ -1,37 +1,72 @@
     let mut snr_hashmap = HashMap::<(u32, u32), (f64, u64)>::new();
-    for node_packet in packets {
-        let node_id = node_packet.id;
-        for neighbor_packet in node_packet.neighbors {
-            let neighbor_id = neighbor_packet.id;
-            // Check if the edge is already in the hashmap
-            let edge_exists = snr_hashmap.contains_key(&as_key(node_id, neighbor_id));
-            if edge_exists {
-                let edge_snr_time = snr_hashmap.get(&as_key(node_id, neighbor_id)).unwrap();
-                let hashmap_timestamp = edge_snr_time.1;
-                // Replace the edge if the new timestamp is more recent (larger)
-                if hashmap_timestamp < neighbor_packet.timestamp {
+    for (_, neighbor_packet) in &neighbors {
+        let node_1 = neighbor_packet.node_id;
+        for neighbor in &neighbor_packet.neighbors {
+            let node_2 = neighbor.node_id;
+            snr_hashmap.insert(
+                as_key(node_1, node_2),
+                (neighbor.snr as f64, neighbor.rx_time as u64),
+            );
+            // If the opposite neighbor is found on a recent packet, we take the most recent timestamp as fact.
+            let opposite_neighbor = get_other_edge_side_packet(node_2, node_1, &neighbors);
+            if opposite_neighbor.is_some() {
+                let opposite_neighbor = opposite_neighbor.unwrap();
+                if neighbor.rx_time > opposite_neighbor.rx_time {
                     snr_hashmap.insert(
-                        as_key(node_id, neighbor_id),
-                        (neighbor_packet.snr, neighbor_packet.timestamp),
+                        as_key(node_1, node_2),
+                        (neighbor.snr as f64, neighbor.rx_time as u64),
+                    );
+                } else {
+                    snr_hashmap.insert(
+                        as_key(node_1, node_2),
+                        (opposite_neighbor.snr as f64, opposite_neighbor.rx_time as u64),
                     );
                 }
             } else {
-                snr_hashmap.insert(
-                    as_key(node_id, neighbor_id),
-                    (neighbor_packet.snr, neighbor_packet.timestamp),
-                );
+                // If the opposite neighbor is not found on a recent packet, we remove the edge from the hashmap.
+                let opt_opposite_node = neighbors.get(&node_2);
+                if opt_opposite_node.is_some() {
+                    let opposite_node = opt_opposite_node.unwrap();
+                    if opposite_node.tx_time > neighbor_packet.tx_time {
+                        if snr_hashmap.contains_key(&as_key(node_1, node_2)) {
+                            snr_hashmap.remove(&as_key(node_1, node_2));
+                        }
+                    }
+                }
             }
         }
     }
     snr_hashmap
 }
 
+// pulls corresponding neighbor from a hashmap of NeighborInfo packets
+pub fn get_other_edge_side_packet(
```

Comment:
Same here, not sure I understand what this function does. Also see `if-let` comment below

Reply:
swapped it with a match as well :)

The purpose of the function is to see if the other side of the edge exists in our `NeighborInfo` packets. I added explanatory comments and renamed the function `check_other_node_agrees`, which may not be any less confusing. Let me know if those two functions make sense now